### PR TITLE
FastAPI Integration

### PR DESCRIPTION
**Pull request by anmarhindi**
Link to main PR:
- https://github.com/code-kern-ai/refinery-gateway/pull/196

### CODE REVIEW

File: src/components/projects/projectId/data-browser/DataBrowser.tsx [hunk 1]
```diff
@@ -64,7 +56,7 @@ export default function DataBrowser() {
         if (!projectId) return;
         if (!searchRequest) return;
         if (searchRequest.offset == 0) return;
-        refetchExtendedRecord({ variables: { projectId: projectId, filterData: JSON.stringify({}), offset: searchRequest.offset, limit: searchRequest.limit } }).then((res) => {
+        searchRecordsExtended(projectId, [], searchRequest.offset, searchRequest.limit, (res) => {
```
Comment (lumburovskalina) [inline src/components/projects/projectId/data-browser/DataBrowser.tsx]: just a note, there is a change in the request here in one other PR for sending the data if there are some conflicts it needs to be changed

File: package.json [hunk 2]
```diff
@@ -41,12 +41,15 @@
...
   },
   "devDependencies": {
+    "@types/jest": "^29.5.12",
     "autoprefixer": "^10.4.15",
+    "jest": "^29.7.0",
     "postcss": "^8.4.28",
-    "tailwindcss": "^3.1.8"
+    "tailwindcss": "^3.1.8",
+    "ts-jest": "^29.1.2",
+    "typescript": "^5.4.2"
```
Comment (lumburovskalina) [inline package.json]: not sure why the typescript was removed from the dependencies, and added only to dev dependencies?

File: src/components/models-download/AddModelDownloadModal.tsx [hunk 3]
```diff
@@ -52,10 +51,8 @@ export default function AddModelDownloadModal() {
         setLineSeparatorIndex(filteredList.findIndex((model: any) => !model.description));
     }, [modelsDownloaded, filteredList, modalAddModel, modelsList, modelName]);
 
-    const [downloadModelMut] = useMutation(MODEL_PROVIDER_DOWNLOAD_MODEL);
-
     const addModel = useCallback(() => {
-        downloadModelMut({ variables: { modelName: modelName } }).then((res) => {
+        modelProviderDownloadModel({ modelName: modelName }, (res) => {
```
Comment (lumburovskalina) [inline src/components/models-download/AddModelDownloadModal.tsx]: we can remove the object and pass it as a string. That would require some changes in the modelProviderDownloadModel (removing {options: {modelName: string}} ) - since we only have one param

File: src/components/models-download/DeleteModelDownloadModal.tsx [hunk 4]
```diff
@@ -16,10 +15,8 @@ export default function DeleteModelDownloadModal() {
 
     const [abortButton, setAbortButton] = useState<ModalButton>(ABORT_BUTTON);
 
-    const [deleteModelDownload] = useMutation(MODEL_PROVIDER_DELETE_MODEL);
-
     const deleteModel = useCallback(() => {
-        deleteModelDownload({ variables: { modelName: modalDeleteModel.modelName } }).then(() => {
+        modelProviderDeleteModel({ modelName: modalDeleteModel.modelName }, (res) => {
```
Comment (lumburovskalina) [inline src/components/models-download/DeleteModelDownloadModal.tsx]: check the same comment for adding a new model

File: src/components/projects/projectId/data-browser/modals/SimilaritySeachModal.tsx [hunk 5]
```diff
@@ -42,18 +42,10 @@ export default function SimilaritySearchModal() {
     const [operatorsDict, setOperatorsDict] = useState<{ [key: string]: string[] }>({});
     const [tooltipsDict, setTooltipsDict] = useState<{ [key: string]: string[] }>({});
 
-    const [refetchSimilarRecords] = useLazyQuery(SEARCH_SIMILAR_RECORDS, { fetchPolicy: 'network-only' });
```
Comment (lumburovskalina) [inline src/components/projects/projectId/data-browser/modals/SimilaritySeachModal.tsx:45]: useLazyQuery is still in the imports

File: src/components/projects/projectId/labeling/sessionId/main-component/LabelingMainComponent.tsx [hunk 6]
```diff
@@ -50,15 +53,7 @@ export default function LabelingMainComponent() {
     const [absoluteWarning, setAbsoluteWarning] = useState(null);
     const [lockedLink, setLockedLink] = useState(false);
 
-    const [refetchHuddleData] = useLazyQuery(REQUEST_HUDDLE_DATA, { fetchPolicy: 'no-cache' });
```
Comment (lumburovskalina) [inline src/components/projects/projectId/labeling/sessionId/main-component/LabelingMainComponent.tsx:53]: useLazyQuery still in the imports

File: src/components/projects/projectId/labeling/sessionId/main-component/LabelingMainComponent.tsx [hunk 7]
```diff
@@ -160,13 +155,15 @@ export default function LabelingMainComponent() {
             return;
         }
         setTimeout(() => {
+            getTokenizedRecord({ recordId: SessionManager.currentRecordId }, (res) => {
+                dispatch(updateRecordRequests('token', res.data.tokenizeRecord));
```
Comment (lumburovskalina) [inline src/components/projects/projectId/labeling/sessionId/main-component/LabelingMainComponent.tsx:155]: getTokenizedRecord  and getRecordByRecordId sent the recordId in a different way

File: src/components/projects/projectId/labeling/sessionId/main-component/NavigationBarTop.tsx [hunk 8]
```diff
@@ -28,12 +27,10 @@ export default function NavigationBarTop(props: NavigationBarTopProps) {
     const displayId = useSelector(selectUserDisplayId);
     const userDisplayRole = useSelector(selectDisplayUserRole);
 
-    const [refetchAvailableLinks] = useLazyQuery(AVAILABLE_LABELING_LINKS, { fetchPolicy: 'no-cache' });
-
     useEffect(() => {
         if (userDisplayRole?.role == UserRole.ENGINEER || !SessionManager.labelingLinkData) return;
         const heuristicId = SessionManager.labelingLinkData.linkType == LabelingLinkType.HEURISTIC ? SessionManager.labelingLinkData.huddleId : null;
-        refetchAvailableLinks({ variables: { projectId: projectId, assumedRole: userDisplayRole?.role, assumedHeuristicId: heuristicId } }).then((result) => {
+        getAvailableLinks(projectId, user?.role, heuristicId, (result) => {
```
Comment (lumburovskalina) [inline src/components/projects/projectId/labeling/sessionId/main-component/NavigationBarTop.tsx]: not sure why this was changed but userDisplayRole?.role is not the same as user?.role, i think this is wrong

File: src/components/projects/projectId/lookup-lists/DeleteLookupListsModal.tsx [hunk 9]
```diff
@@ -21,19 +20,11 @@ export default function DeleteLookupListsModal(props: DeleteLookupListsModalProp
...
         checkedLookupLists.forEach((checked, index) => {
             if (checked) {
                 const lookupList = lookupLists[index];
-                deleteLookupListMut({
-                    variables: {
-                        projectId: projectId,
-                        knowledgeBaseId: lookupList.id
-                    }
-                }).then((res) => {
+                deleteKnowledgeBase(projectId, lookupList.id, (res) => {
```
Comment (lumburovskalina) [inline src/components/projects/projectId/lookup-lists/DeleteLookupListsModal.tsx:27]: if there is time, delete and create Knowledgebase should be renamed to delete and create Lookuplist since we use those names now.

File: src/components/projects/projectId/lookup-lists/lookupListId/PasteLookupListModal.tsx [hunk 10]
```diff
@@ -17,10 +16,13 @@ export default function PasteLookupListModal() {
...
 
-    const [pasteLookupListMut] = useMutation(PASTE_TERM);
-
     const pasteLookupList = useCallback(() => {
-        pasteLookupListMut({ variables: { projectId: projectId, knowledgeBaseId: router.query.lookupListId, values: inputArea, split: inputSplit, delete: false } }).then((res) => {
+        pasteKnowledgeTerms(projectId, {
+            knowledgeBaseId: router.query.lookupListId,
+            values: inputArea,
+            split: inputSplit,
+            delete: true
```
Comment (lumburovskalina) [inline src/components/projects/projectId/lookup-lists/lookupListId/PasteLookupListModal.tsx]: delete should be false

File: src/components/projects/projectId/lookup-lists/lookupListId/Terms.tsx [hunk 11]
```diff
@@ -1,11 +1,10 @@
 import { selectProjectId } from "@/src/reduxStore/states/project";
-import { ADD_TERM_TO_LOOKUP_LIST, BLACKLIST_TERM, REMOVE_TERM, UPDATE_TERM } from "@/src/services/gql/mutations/lookup-lists";
+import { deleteTerm as dlt, blacklistTerm as blt, addTermToKnowledgeBase as tkb, updateTerm } from "@/src/services/base/lookup-lists";
```
Comment (lumburovskalina) [inline src/components/projects/projectId/lookup-lists/lookupListId/Terms.tsx]: i think the naming is not clear, if i take a look at the code in few weeks i wouldn't know what's blt and tkb. The names aren't that long so i would go with a full name

File: src/components/projects/projectId/settings/ProjectMetaData.tsx [hunk 12]
```diff
@@ -37,7 +33,7 @@ export default function ProjectMetaData() {
     }
 
     function deleteProject() {
-        deleteProjectMut({ variables: { projectId: project.id } }).then(() => {
+        dlp(project.id, (res) => {
```
Comment (lumburovskalina) [inline src/components/projects/projectId/settings/ProjectMetaData.tsx]: same as above, not really clear (maybe it's only personal preference)

File: src/components/projects/projectId/settings/labeling-tasks/ChangeColorModal.tsx [hunk 13]
```diff
@@ -2,17 +2,16 @@ import Modal from "@/src/components/shared/modal/Modal";
...
 import { ModalEnum } from "@/src/types/shared/modal";
 import { LabelHelper } from "@/src/util/classes/label-helper";
 import { TOOLTIPS_DICT } from "@/src/util/tooltip-constants";
 import { jsonCopy } from "@/submodules/javascript-functions/general";
-import { useMutation } from "@apollo/client";
 import { Tooltip } from "@nextui-org/react";
 import { IconPencil } from "@tabler/icons-react";
 import { useEffect, useState } from "react";
 import { useDispatch, useSelector } from "react-redux";
+import { updateLabelColor as ulc, updateLabelHotkey } from "@/src/services/base/labeling";
```
Comment (lumburovskalina) [inline src/components/projects/projectId/settings/labeling-tasks/ChangeColorModal.tsx]: same as naming above

File: src/components/projects/projectId/settings/labeling-tasks/DeleteLabelModal.tsx [hunk 14]
```diff
@@ -19,11 +18,9 @@ export default function DeleteLabelModal() {
 
     const [abortButton, setAbortButton] = useState<ModalButton>(ABORT_BUTTON);
 
-    const [deleteLabelMut] = useMutation(DELETE_LABEL);
-
     const deleteLabel = useCallback(() => {
         LabelHelper.removeLabel(modalDeleteLabel.taskId, modalDeleteLabel.label.color.name);
-        deleteLabelMut({ variables: { projectId: projectId, labelId: modalDeleteLabel.label.id } }).then(() => {
+        dl(projectId, modalDeleteLabel.label.id, (res) => {
```
Comment (lumburovskalina) [inline src/components/projects/projectId/settings/labeling-tasks/DeleteLabelModal.tsx]: same as naming above

File: src/components/projects/projectId/settings/labeling-tasks/DeleteLabelingTaskModal.tsx [hunk 15]
```diff
@@ -18,10 +17,8 @@ export default function DeleteLabelingTaskModal() {
 
     const [abortButton, setAbortButton] = useState<ModalButton>(ABORT_BUTTON);
 
-    const [deleteLabelingTaskMut] = useMutation(DELETE_LABELING_TASK);
-
     const deleteLabelingTask = useCallback(() => {
-        deleteLabelingTaskMut({ variables: { projectId: projectId, labelingTaskId: modalDeleteLabelingTask.taskId } }).then((res) => {
+        dlt(projectId, modalDeleteLabelingTask.taskId, (res) => {
```
Comment (lumburovskalina) [inline src/components/projects/projectId/settings/labeling-tasks/DeleteLabelingTaskModal.tsx]: same as naming above

File: src/components/shared/comments/CommentsMainSection.tsx [hunk 16]
```diff
@@ -7,12 +7,11 @@ import { useCallback, useEffect, useState } from "react";
 import DisplayComments from "./DisplayComments";
 import { useSelector } from "react-redux";
 import { selectAllUsers, selectComments } from "@/src/reduxStore/states/general";
-import { CREATE_COMMENT, DELETE_COMMENT, UPDATE_COMMENT } from "@/src/services/gql/mutations/projects";
-import { useMutation } from "@apollo/client";
 import { selectProjectId } from "@/src/reduxStore/states/project";
 import { CommentCreation } from "./CommentCreation";
 import { convertTypeToKey } from "@/src/util/shared/comments-helper";
 import { CommentDataManager } from "@/src/util/classes/comments";
+import { createComment, deleteComment as del, updateComment as upd } from "@/src/services/base/comment";
```
Comment (lumburovskalina) [inline src/components/shared/comments/CommentsMainSection.tsx]: same as above

File: src/components/shared/modal/Modal.tsx [hunk 17]
```diff
@@ -3,7 +3,8 @@ import { Fragment } from 'react'
 import { Dialog, Transition } from '@headlessui/react'
 import { isStringTrue } from '@/submodules/javascript-functions/general';
 import { useDispatch, useSelector } from 'react-redux';
-import { Modal, closeModal, initModal, openModal, selectModal, selectAllOpenModals } from '@/src/reduxStore/states/modal';
+import { closeModal, initModal, openModal, selectModal, selectAllOpenModals } from '@/src/reduxStore/states/modal';
+import type { Modal } from '@/src/reduxStore/states/modal';
```
Comment (lumburovskalina) [inline src/components/shared/modal/Modal.tsx:7]: why was this changed?

File: src/components/shared/upload/Upload.tsx [hunk 18]
```diff
@@ -4,23 +4,21 @@ import UploadField from "./helper-components/UploadField";
...
 import { extendAllProjects, removeFromAllProjectsById, selectAllProjects, selectProjectId } from "@/src/reduxStore/states/project";
-import { GET_UPLOAD_CREDENTIALS_AND_ID, GET_UPLOAD_TASK_BY_TASK_ID } from "@/src/services/gql/queries/projects";
 import { UPLOAD_TOKENIZERS } from "@/src/util/constants";
 import { UploadHelper, ZIP_TYPE } from "@/src/util/classes/upload-helper";
 import CryptedField from "../crypted-field/CryptedField";
 import { closeModal } from "@/src/reduxStore/states/modal";
 import { ModalEnum } from "@/src/types/shared/modal";
 import Dropdown2 from "@/submodules/react-components/components/Dropdown2";
 import { useWebsocket } from "@/src/services/base/web-sockets/useWebsocket";
+import { getUploadCredentialsAndId, getUploadTaskById, deleteProject as dlp, createProject as crp, updateProjectTokenizer, updateProjectStatus } from "@/src/services/base/project";
```
Comment (lumburovskalina) [inline src/components/shared/upload/Upload.tsx]: same as above with naming

File: src/services/base/embedding.ts [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+export function getAllTokenizerOptions(onResult: (result: any) => void) {
+    const finalUrl = `${embeddingEndpoint}/language-models`;
+
+    jsonFetchWrapper(finalUrl, FetchType.GET, onResult);
+}
+
+export function getEmbeddings(projectId: string, onResult: (result: any) => void) {
+    const finalUrl = `${embeddingEndpoint}/embeddings-by-project?project_id=${projectId}`;
```
Comment (lumburovskalina) [inline src/services/base/embedding.ts]: why not part of the url? in most cases the projectId is part of the url

File: src/services/base/labeling.ts [hunk 20]
```diff
@@ -0,0 +1,164 @@
+import { FetchType, jsonFetchWrapper } from "@/submodules/javascript-functions/basic-fetch";
+import { BACKEND_BASE_URI } from "./_settings";
+import { LabelingLinkType } from "@/src/types/components/projects/projectId/labeling/labeling-main-component";
+import { convertCamelToSnakeCase } from "@/submodules/javascript-functions/case-types-parser";
+
+export const labelingEndpoint = `${BACKEND_BASE_URI}/api/v1/labeling`;
+
+export function getAvailableLinks(projectId: string, assumedRole: string, assumedHeuristicId: string, onResult: (result: any) => void) {
+    const finalUrl = `${labelingEndpoint}/${projectId}/available-links`;
+    jsonFetchWrapper(finalUrl, FetchType.POST, onResult, JSON.stringify({ assumedRole, assumedHeuristicId }));
```
Comment (lumburovskalina) [inline src/services/base/labeling.ts:10]: convertCamelToSnakeCase missing ?

File: src/services/base/labeling.ts [hunk 21]
```diff
@@ -0,0 +1,164 @@
...
+export function getTokenizedRecord(options: {
+    recordId: string
+}, onResult: (result: any) => void) {
+    const finalUrl = `${labelingEndpoint}/tokenized-record`;
+    jsonFetchWrapper(finalUrl, FetchType.POST, onResult, JSON.stringify(convertCamelToSnakeCase(options)));
+}
+
+export function deleteRecordLabelAssociationByIds(projectId: string, recordId: string, associationIds: string[], onResult: (result: any) => void) {
+    const finalUrl = `${labelingEndpoint}/${projectId}/record-label-association-by-ids`;
+    jsonFetchWrapper(finalUrl, FetchType.DELETE, onResult, JSON.stringify({ recordId, associationIds }));
```
Comment (lumburovskalina) [inline src/services/base/labeling.ts:29]: convertCamelToSnakeCase missing ?

File: src/services/base/labeling.ts [hunk 22]
```diff
@@ -0,0 +1,164 @@
...
+export function generateAccessLink(projectId: string, options: {
+    type: string, id: string,
+}, onResult: (result: any) => void) {
+    const finalUrl = `${labelingEndpoint}/${projectId}/generate-access-link`;
+    jsonFetchWrapper(finalUrl, FetchType.POST, onResult, JSON.stringify(convertCamelToSnakeCase(options)));
+}
+
+export function removeAccessLink(projectId: string, linkId: string, onResult: (result: any) => void) {
+    const finalUrl = `${labelingEndpoint}/${projectId}/remove-access-link`;
+    jsonFetchWrapper(finalUrl, FetchType.DELETE, onResult, JSON.stringify({ "value": linkId }));
```
Comment (lumburovskalina) [inline src/services/base/labeling.ts:53]: convertCamelToSnakeCase missing ? if yes check the other occurrences in the file

File: src/services/base/lookup-lists.ts [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+
+export function addTermToKnowledgeBase(projectId: string, options: {
+    value: string, comment: string, knowledgeBaseId: string | string[]
+}, onResult: (result: any) => void) {
+    const finalUrl = `${lookupListsEndpoint}/${projectId}/add-term-to-knowledge-base`;
+    jsonFetchWrapper(finalUrl, FetchType.POST, onResult, JSON.stringify(convertCamelToSnakeCase(options)));
+}
+
+export function deleteTerm(projectId: string, termId: string, onResult: (result: any) => void) {
+    const finalUrl = `${lookupListsEndpoint}/${projectId}/delete-term/${termId}`;
```
Comment (lumburovskalina) [inline src/services/base/lookup-lists.ts]: i would switch places for delete-term and termId, the same for blacklistTerm so they differentiate

File: src/services/base/notification.ts [hunk 24]
```diff
@@ -0,0 +1,9 @@
+import { FetchType, jsonFetchWrapper } from "@/submodules/javascript-functions/basic-fetch";
+import { BACKEND_BASE_URI } from "./_settings";
+
+export const notificationEndpoint = `${BACKEND_BASE_URI}/api/v1/notification`;
+
+export function getNotificationsByUser(onResult: (result: any) => void) {
+    const finalUrl = `${notificationEndpoint}`;
+    jsonFetchWrapper(finalUrl, FetchType.GET, onResult);
+}
```
Comment (lumburovskalina) [inline src/services/base/notification.ts]: the query for getting all notifications can be part of the notifications route instead of being part of the projects route

File: src/services/base/organization.ts [hunk 25]
```diff
@@ -0,0 +1,64 @@
...
+    const finalUrl = `${organizationEndpoint}/overview-stats`;
+    jsonFetchWrapper(finalUrl, FetchType.GET, onResult);
+}
+
+export function getAllActiveAdminMessages(onResult: (result: any) => void) {
+    const finalUrl = `${organizationEndpoint}/all-active-admin-messages`;
+    jsonFetchWrapper(finalUrl, FetchType.GET, onResult);
+}
+
+export function getAllUsersWithRecordCount(projectId: string, onResult: (result: any) => void) {
```
Comment (lumburovskalina) [inline src/services/base/organization.ts:32]: this is used for the data browser, i think we can moved it there, but nothing critical, can also stay here

File: src/services/base/labeling.ts [hunk 26]
```diff
@@ -0,0 +1,164 @@
...
+export function removeGoldStar(projectId: string, recordId: string, labelingTaskId: string, onResult: (result: any) => void) {
+    const finalUrl = `${labelingEndpoint}/${projectId}/remove-gold-star`;
+    const body = {
+        recordId,
+        labelingTaskId
+    }
+    jsonFetchWrapper(finalUrl, FetchType.POST, onResult, JSON.stringify(convertCamelToSnakeCase(body)));
+}
+
+export function updateLabelingTask(projectId: string, options: {
```
Comment (lumburovskalina) [inline src/services/base/labeling.ts]: everything from here to the end should go in a separate file in my opinion since they are not part of the labeling. They are part of the project settings but i think the logic can be split into separate file so it's not to much code in one file

File: src/util/components/models-downloaded/models-downloaded-helper.ts [hunk 27]
```diff
@@ -11,6 +11,7 @@ export function postProcessingModelsDownload(modelsDownloaded: ModelsDownloaded[
 export function postProcessingZeroShotEncoders(zeroShotModels: ModelsDownloaded[], encoders: ModelsDownloaded[]): any {
     const prepareModelsList = encoders.filter((el: any) =>
         el.configString != 'bag-of-characters' && el.configString != 'bag-of-words' && el.configString != 'tf-idf');
-    zeroShotModels.sort((a, b) => a.prio - b.prio);
-    return prepareModelsList.concat(zeroShotModels);
+    const zeroShotModelsCopy = [...zeroShotModels];
```
Comment (lumburovskalina) [inline src/util/components/models-downloaded/models-downloaded-helper.ts:14]: not sure why this was changed, but i posted a screenshot that the ordering of the zero-shot models doesn't work as expected, maybe related to this?

File: src/components/projects/projectId/labeling/sessionId/main-component/LabelingMainComponent.tsx (hunk 7 above)
Comment (lumburovskalina) [reply src/components/projects/projectId/labeling/sessionId/main-component/LabelingMainComponent.tsx:155]: won't be changed, personal preference

File: src/services/base/labeling.ts (hunk 22 above)
Comment (anmarhindi) [reply src/services/base/labeling.ts:53]: in this case its not needed as the key is `value`

File: src/services/base/labeling.ts (hunk 20 above)
Comment (anmarhindi) [reply src/services/base/labeling.ts:10]: can be ignored as its fetched directly from the json body

File: src/services/base/labeling.ts (hunk 21 above)
Comment (anmarhindi) [reply src/services/base/labeling.ts:29]: can be ignored as its fetched directly from the json body

File: src/components/projects/projectId/lookup-lists/DeleteLookupListsModal.tsx (hunk 9 above)
Comment (anmarhindi) [reply src/components/projects/projectId/lookup-lists/DeleteLookupListsModal.tsx:27]: KnowledgeBase is used quite a bit in the backend, will be time consuming. My recommendation would be to ignore for now in favor for next release (or even later, as non prio but can consume time)

Comment (lumburovskalina) [reply src/components/projects/projectId/lookup-lists/DeleteLookupListsModal.tsx:27]: yes i agree, i am resolving the issue since it's non prio

File: src/components/shared/modal/Modal.tsx (hunk 17 above)
Comment (anmarhindi) [reply src/components/shared/modal/Modal.tsx:7]: build breaks if Modal not imported as type due to existing Modal function

Comment (lumburovskalina) [reply src/components/shared/modal/Modal.tsx:7]: okay , i will resolve the conversation

File: src/services/base/embedding.ts (hunk 19 above)
Comment (anmarhindi) [reply src/services/base/embedding.ts:26]: @JWittmeyer this request also seems to break in pages.

File: src/util/components/models-downloaded/models-downloaded-helper.ts (hunk 27 above)
Comment (lumburovskalina) [reply src/util/components/models-downloaded/models-downloaded-helper.ts:14]: not related, resolving the conversation

File: src/services/base/organization.ts (hunk 25 above)
Comment (lumburovskalina) [reply src/services/base/organization.ts:32]: i checked the BE, it's part of the organization manager, so we can leave it in the organization route

File: src/components/models-download/AddModelDownloadModal.tsx [hunk 28]
```diff
@@ -82,7 +79,7 @@ export default function AddModelDownloadModal() {
                     <span className="card-title mb-0 label-text flex"><span className="cursor-help underline filtersUnderline">Name</span></span>
                 </Tooltip>
                 <Dropdown2 options={filteredList && filteredList} useDifferentTextColor={colorDownloadedModels} differentTextColor="green" valuePropertyPath="configString"
-                    hasSearchBar={true} dropdownItemsClasses="max-h-96 overflow-y-auto" ignoreDisabledForSearch={true}
```
Comment (lumburovskalina) [inline src/components/models-download/AddModelDownloadModal.tsx]: why was ignoreDisabledForSearch removed?
I added it intentionally

File: src/components/projects/projectId/heuristics/modals/AddZeroShotModal.tsx [hunk 29]
```diff
@@ -160,7 +155,7 @@ export default function AddZeroShotModal() {
                 </div>
             </Tooltip>
             <Dropdown2 options={filteredList && filteredList} hasSearchBar={true} optionsHaveLink={true} optionsHaveHoverBox={true} valuePropertyPath="configString"
-                useDifferentTextColor={colorDownloadedModels} differentTextColor="green" ignoreDisabledForSearch={true}
+                useDifferentTextColor={colorDownloadedModels} differentTextColor="green"
                 linkList={filteredList && filteredList.map(model => model.link)}
```
Comment (lumburovskalina) [inline src/components/projects/projectId/heuristics/modals/AddZeroShotModal.tsx]: see the comment above

File: src/util/components/projects/projectId/labeling/labeling-helper.ts [hunk 30]
```diff
@@ -158,8 +158,10 @@ export function collectSelectionData(attributeId: string, currentSelection: Curr
 export function getTokenData(attributeId: string, attributes: Attribute[], recordRequests: any): any {
     if (!attributes) return null;
     if (attributeId == FULL_RECORD_ID) return null;
-    for (const att of recordRequests.token.attributes) {
-        if (att.attributeId == attributeId) return att;
+    if (recordRequests.token.length > 0) {
```
Comment (lumburovskalina) [inline src/util/components/projects/projectId/labeling/labeling-helper.ts]: i think this is wrong, recordRequests.token is not an array

File: src/components/models-download/AddModelDownloadModal.tsx (hunk 28 above)
Comment (anmarhindi) [reply src/components/models-download/AddModelDownloadModal.tsx]: The ignoreDisabledForSearch was causing a build error. Seems like an non-existing prop.

File: src/components/projects/projectId/heuristics/modals/AddZeroShotModal.tsx (hunk 29 above)
Comment (anmarhindi) [reply src/components/projects/projectId/heuristics/modals/AddZeroShotModal.tsx]: Causing build issue.

File: src/components/models-download/AddModelDownloadModal.tsx (hunk 28 above)
Comment (lumburovskalina) [reply src/components/models-download/AddModelDownloadModal.tsx]: have you merged the submodules? it's part of the react-components (new propery to the Dropdown2)

Comment (anmarhindi) [reply src/components/models-download/AddModelDownloadModal.tsx]: Oops! Oversight on my side! I will re-add them now and test it again.

File: src/util/components/projects/projectId/labeling/labeling-helper.ts [hunk 31]
```diff
@@ -158,10 +158,8 @@ export function collectSelectionData(attributeId: string, currentSelection: Curr
 export function getTokenData(attributeId: string, attributes: Attribute[], recordRequests: any): any {
     if (!attributes) return null;
     if (attributeId == FULL_RECORD_ID) return null;
-    if (recordRequests.token.length > 0) {
-        for (const att of recordRequests.token.attributes) {
-            if (att.attributeId == attributeId) return att;
-        }
+    for (const att of recordRequests.token.attributes) {
+        if (att.attributeId == attributeId) return att;
```
Comment (anmarhindi) [inline src/util/components/projects/projectId/labeling/labeling-helper.ts]: Screenshot 2024-04-15 at 15 35 52

Causing issues when changes were reverted.

File: src/services/base/labeling.ts [hunk 32]
```diff
@@ -67,3 +67,18 @@ export function addClassificationLabels(projectId: string, recordId: string, lab
...
+        labelingTaskId,
+        tokenStartIndex,
+        tokenEndIndex,
+        value,
+        labelId,
+        asGoldStar,
+        sourceId
+    }
+    jsonFetchWrapper(finalUrl, FetchType.POST, onResult, JSON.stringify(convertCamelToSnakeCase(body)));
+}
```
Comment (anmarhindi) [inline src/services/base/labeling.ts]: Screenshot 2024-04-15 at 15 39 26

Currently not allowing me to add highlighted words, seems its stuck in a certain position.

File: src/util/components/projects/projectId/labeling/labeling-helper.ts (hunk 31 above)
Comment (lumburovskalina) [reply src/util/components/projects/projectId/labeling/labeling-helper.ts]: then something else causes the issue, but i tested this on thursday and adding information extraction was working without issues. Still even this is removes, the values from the BE are wrong

**Review by lumburovskalina — APPROVED** [review]
